File: java/src/test/java/problems101_200/PascalsTriangleTest.java
package problems101_200;

import java.util.Arrays;
import java.util.stream.Collectors;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class PascalsTriangleTest extends SampledTest {
    @Test
    public void test() {
        for (PascalsTriangle solution : PascalsTriangle.values()) {
            for (PascalsTriangleSample sample : getSamples(PascalsTriangleSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.generate(sample.input.numRows))
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .map(ints ->
                                Arrays
                                    .stream(ints)
                                    .boxed()
                                    .collect(Collectors.toList())
                            ).collect(Collectors.toList())
                    );
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/PascalsTriangleTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class PascalsTriangleTest : SampledTest() {
    @Test
    fun test() =
        PascalsTriangle.entries.forEach { solution ->
            getSamples(Array<PascalsTriangleSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.generate(input.numRows))
                        .containsExactlyElementsIn(output.map { it.toList() })
                }
        }
}
